cmd/thanos/query.go

```diff
@@ -529,12 +568,72 @@ func runQuery(
 		})
 	}
 
-	grpcProbe := prober.NewGRPC()
-	httpProbe := prober.NewHTTP()
-	statusProber := prober.Combine(
-		httpProbe,
-		grpcProbe,
-		prober.NewInstrumentation(comp, logger, extprom.WrapRegistererWithPrefix("thanos_", reg)),
+	var (
+		allClients = func() []store.Client {
+			var get []store.Client
+			for _, ss := range storeSets {
+				get = append(get, ss.Get()...)
+			}
```

Comment:
We have this loop in all the 4 variables below, I tried to have this once globally and then just retuning the variables inside these func(), but it's weird to see that it was not working as expected (e2e tests were also failing) and I have to change and make it repetitive in this [commit](https://github.com/thanos-io/thanos/pull/4389/commits/672b9d7810f36e619d34e52f679cbd2670a92a3d) after which all tests were happy.